packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java

```diff
@@ -24,14 +30,108 @@
 
 public class FlutterWebView implements PlatformView, MethodCallHandler {
   private static final String JS_CHANNEL_NAMES_FIELD = "javascriptChannelNames";
-  private final WebView webView;
+  private final InputAwareWebView webView;
   private final MethodChannel methodChannel;
   private final FlutterWebViewClient flutterWebViewClient;
   private final Handler platformThreadHandler;
+  private final View flutterView;
+
+  private static class InputAwareWebView extends WebView {
+    private static final String TAG = "InputAwareWebView";
+    private final View flutterView;
+
+    InputAwareWebView(Context context, View flutterView) {
+      super(context);
+      this.flutterView = flutterView;
+    }
+
+    private View threadedInputConnectionProxyView;
+
+    private ThreadedInputConnectionProxyAdapterView proxyAdapterView;
+
+    @Override
+    public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
+      if (proxyAdapterView == null) {
+        // No proxy adapter set, so we use the default implementation.
+        return super.onCreateInputConnection(outAttrs);
+      }
+
+      if (!proxyAdapterView.isTriggerDelayed()) {
+        // Currently on the IME thread. Delegate to the superclass.
+        return super.onCreateInputConnection(outAttrs);
+      }
+
+      InputConnection result = super.onCreateInputConnection(outAttrs);
+      if (result != null) {
+        // This should never happen.
+        Log.wtf(TAG, "Failed unexpectedly creating a webview input connection.");
+      }
+
+      final InputMethodManager imm =
+          (InputMethodManager) getContext().getSystemService(INPUT_METHOD_SERVICE);
+      proxyAdapterView.requestFocus();
+      final View containerView = this;
+
+      // This is the crucial trick that gets the InputConnection creation to happen on the correct
+      // thread. https://cs.chromium.org/chromium/src/content/public/android/java/src/org/chromium/content/browser/input/ThreadedInputConnectionFactory.java?l=169.
```

Comment:
nit, let's pin the link to a specific version in case it moves e.g: https://cs.chromium.org/chromium/src/content/public/android/java/src/org/chromium/content/browser/input/ThreadedInputConnectionFactory.java?l=169&rcl=f0698ee3e4483fad5b0c34159276f71cfaf81f3a